python/lsst/meas/deblender/deblend.py

```diff
@@ -674,50 +677,59 @@ def __init__(self, schema, peakSchema=None, **kwargs):
             err = "Currently the only supported translationMethod is 'default', you entered '{0}'"
-            raise ValueError(err.format(self.config.translationMethod))
-        else:
-            Translation = deblender.operators.TxyTranslation
-        if self.config.multipleComponents:
-            components = ["bulge", "disk"]
-        else:
-            components = None
-        if self.config.updateMorphologyFirst:
-            updateOrder = [1,0]
-        else:
-            updateOrder = [0,1]
-        if np.isnan(self.config.l0Thresh):
-            l0Thresh = None
-        else:
-            l0Thresh = self.config.l0Thresh
-        if np.isnan(self.config.l1Thresh):
-            l1Thresh = None
-        else:
-            l1Thresh = self.config.l1Thresh
-
-        multiband_plugin = plugins.DeblenderPlugin(plugins.buildMultibandTemplates,
-                                                   max_iter=self.config.maxIter,
-                                                   useWeights=self.config.useWeights,
-                                                   e_rel=self.config.relativeError,
-                                                   e_abs=self.config.absoluteError,
-                                                   fit_positions=self.config.fitPositions,
-                                                   txy_diff=self.config.diffImgShift,
-                                                   txy_thresh=self.config.minTranslation,
-                                                   max_shift=self.config.maxTranslation,
-                                                   txy_wait=self.config.requireTranslation,
-                                                   txy_skip=self.config.skipTranslation,
-                                                   Translation=Translation,
-                                                   components=components,
-                                                   constraints=self.config.constraints,
-                                                   l0_thresh=l0Thresh,
-                                                   l1_thresh=l1Thresh,
-                                                   useStrictMonotonicity=self.config.useStrictMonotonicity,
-                                                   monotonicUseNearest=self.config.useNearestMonotonic,
-                                                   smoothness=self.config.smoothness,
-                                                   slack=self.config.stepSlack,
-                                                   usePsf=self.config.usePsfConvolution,
-                                                   update_order=updateOrder)
+            raise NotImplementedError(err.format(self.config.translationMethod))
+
+        # If the default constraints are not used, set the constraints for
+        # all of the sources
+        constraints = None
+        if (sorted(self.config.constraints) != ['+', '1', 'M', 'S']
```

Comment:
Ah, so it should be a list above? Or at least specify that ordering doesn't matter.